src/reducers.js

```diff
@@ -63,23 +64,37 @@ const perPage = (state = PER_PAGE) => {
   return state
 }
 
-// const errorReport = (state = '', action) => {
-//   switch (action.type) {
-//     case REPORT_ERROR:
-//       console.log(`${action.errorCode}: ${action.errorMessage}`)
-//       return `${action.errorCode}: ${action.errorMessage}`
-//     default:
-//       return state
-//   }
-
-// }
-
-const rootReducer = combineReducers({
+const appReducer = combineReducers({
   userData,
   currentPage,
   allUsers,
   pageUsers,
   perPage
 })
 
+// Async actions
+// ---------------------------------------------------------------------------
+const login = (email, password) => {
+  return firebase.auth().signInWithEmailAndPassword(email, password)
+    .then((user) => {
+      firebase.auth().onAuthStateChanged((fbuser) => {
+        return fbuser ? fbuser.uid : null
+      })
+    })
+}
+
+const rootReducer = (state, action) => {
+  console.log(state, action)
+  if (action.type === LOG_IN) {
+    return loop(
+      state,
+      Cmd.run(login, {
+        successActionCreator: setAUser,
+        failActionCreator: reportError,
+        args: [state.userData.email, state.userData.password]
+      }))
+  }
```

Comment:
So here is where the redux-loop magic comes in. We run `loop()` when we want to kick off an async action and then we define the two actions that will be called on success or failure of the async action and the arguments that will be passed to that async action.